### PR TITLE
Lazy Deposit All Module Accounts During EndBlock

### PR DESCRIPTION
**Pull request by BrandonWeng**
## Describe your changes and provide context
Main changes in this PR: 
https://github.com/sei-protocol/sei-cosmos/pull/58 

We are moving the module deposits to the end of the block and transferring them all at once. This is to prevent the gas fee deposit from being a single source of a bottleneck when it comes to concurrency. 

## Testing performed to validate your change
Ran 3 iterations of load tests with 5 rounds, and 400 orders per block 
image

image

### CODE REVIEW

**Review by codchen — APPROVED** [review]

File: app/app.go [hunk 1]
```diff
@@ -1007,6 +1008,7 @@ func (app *App) ProcessBlockConcurrent(
 		return txResults
 	}
 
+	ctx.Logger().Info(fmt.Sprintf("Processing Tx Count=%d", len(txs)))
```
Comment (philipsu522) [inline app/app.go]: nit - might be good to make this debug & a metric instead

File: app/app.go [hunk 2]
```diff
@@ -1032,6 +1034,7 @@ func (app *App) ProcessBlockConcurrent(
 	// Gather Results and store it based on txIndex and read results from channel
 	// Concurrent results may be in different order than the original txIndex
 	txResultsMap := map[int]*abci.ExecTxResult{}
+	ctx.Logger().Info("Waiting for TXs to return")
```
Comment (philipsu522) [inline app/app.go]: maybe set this to debug?

File: x/oracle/ante.go [hunk 3]
```diff
@@ -47,6 +49,31 @@ func (spd SpammingPreventionDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, si
 	return next(ctx, tx, simulate)
 }
 
+func (spd SpammingPreventionDecorator) AnteDeps(txDeps []sdkacltypes.AccessOperation, tx sdk.Tx, next sdk.AnteDepGenerator) (newTxDeps []sdkacltypes.AccessOperation, err error) {
```
Comment (philipsu522) [inline x/oracle/ante.go:52]: are these AnteDeps functions going to look similar across all antes (i.e. oracle, gasless)? If so, maybe it'll make sense to have this somewhere centralized. If not, ffti

**Review by philipsu522 — APPROVED** [review]

File: x/oracle/ante.go (hunk 3 above)
Comment (BrandonWeng) [reply x/oracle/ante.go:52]: Tony did that in his PR - I'm going to remove my changes and rebase on his

**Pull request merged**